runtime/nvqir/NVQIR.cpp

```diff
@@ -635,6 +635,77 @@ void __quantum__qis__custom_unitary(std::complex<double> *unitary,
       unitaryMatrix, ctrlsVec, tgtsVec, name);
 }
 
+void __quantum__qis__apply_kraus_channel(std::int64_t krausChannelKey,
+                                         double *params, std::size_t numParams,
+                                         Array *qubits) {
+
+  auto *ctx = nvqir::getCircuitSimulatorInternal()->getExecutionContext();
+  if (!ctx)
+    return;
+
+  auto *noise = ctx->noiseModel;
+  if (!noise)
+    return;
+
+  std::vector<double> paramVec(params, params + numParams);
+  auto channel = noise->get_channel(krausChannelKey, paramVec);
+  nvqir::getCircuitSimulatorInternal()->applyNoise(channel,
+                                                   arrayToVectorSizeT(qubits));
+}
+
+void __quantum__qis__apply_kraus_channel_generalized(
+    std::int64_t krausChannelKey, std::size_t numSpans, std::size_t numParams,
+    std::size_t numTargets, ...) {
+  struct basic_span {
+    double *_0;
+    std::size_t _1;
```

Comment:
Suggest more meaningful names.
```suggestion
    double *ptr;
    std::size_t len;
```